### PR TITLE
Update test suite, update Solidity version for compiling test contracts, remove deprecated JSON-RPC methods

### LINKED ISSUE #3284: Recompile test contracts with new Solidity version: v0.8.25
### What was wrong?<br>

* New Solidity version `v0.8.25` released and test contracts need to be recompiled with this version.<br>

### How can it be fixed?<br>
* From within the `web3/_utils/contract_sources` directory, run: `python compile_contracts.py -v {new_solidity_version_number}` to compile the test contracts with the new Solidity version.<br>
* Re-generate the integration test fixture after running the above script to test the integration tests against the updated contract bytecodes.<br><br>See [relevant documentation](https://ift.tt/r75lQwa) for more details.

### LINKED ISSUE #3405: Recompile test contracts with new Solidity version: v0.8.26
### What was wrong?<br>

* New Solidity version `v0.8.26` released and test contracts need to be recompiled with this version.<br>

### How can it be fixed?<br>
* From within the `web3/_utils/contract_sources` directory, run: `python compile_contracts.py -v {new_solidity_version_number}` to compile the test contracts with the new Solidity version.<br>
* Re-generate the integration test fixture after running the above script to test the integration tests against the updated contract bytecodes.<br><br>See [relevant documentation](https://ift.tt/A5fbjFY) for more details.

### PR DESCRIPTION
**Pull request by fselmo**
### What was wrong?

This PR was meant to update Solidity contract versions but it revealed that `geth --dev` was broken. Geth fixed this and this PR brings those changes in (`v1.14.3`). Updated geth no longer has support for `eth_mining`, `eth_hashrate`, `eth_submitWork`, `eth_getWork`, `eth_coinbase`. This PR gets our test suite updated with all of these changes as well as removes support for those methods altogether. We will have to issue a backport PR for `v6` to deprecate all of these.

- update py-geth
- update geth version to `v1.14.3`
- update `eth-tester[py-evm]` to `0.10.0b.4`
- closes #3405 
- remove support for `eth_mining`, `eth_hashrate`, `eth_submitHashrate`, `eth_submitWork`, `eth_getWork`, `eth_coinbase`
- remove support for `pending` block

---

The goal here was to compile test contracts with Solidity ``0.8.25`` which broke because we needed to update geth. This PR was open for a while and `0.8.26` came out as well. [This build](https://app.circleci.com/pipelines/github/ethereum/web3.py/6756) has everything working with `0.8.25` and then the latest version should use `0.8.26`.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

`0_o`

### CODE REVIEW

**Comment by fselmo** [comment]
fyi: I updated my gpg key so I had to squash all old commits into one large commit that was signed by the new gpg key. I would've used a bit more separation of concerns in the commits / messages otherwise.

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,14 +5,14 @@ version: 2.1
 parameters:
   geth_version:
     # update default value when updating geth integration test fixture
-    default: "v1.13.14"
+    default: "v1.14.5"
     type: string
   pygeth_version:
     # update default value when updating geth integration test fixture
-    default: "4.3.0"
+    default: "5.0.0b1"
```
Comment (reedsa) [inline .circleci/config.yml:12]: Is this meant to hold until py-geth v5 is released?

**Review by reedsa — COMMENTED** [review]
Looks good so far. Not sure what happened with the latest run in CI but I'll have another look with the next `py-geth` release.

**Comment by fselmo** [comment]
> Not sure what happened with the latest run in CI

Circle CI was down EOD yesterday. I meant to re-run today. 

> I'll have another look with the next `py-geth` release

I don't think we plan on releasing it while keeping this PR open. This is going in *web3.py* `main` which is also in beta so I think we should merge it as is and just make sure to update to stable before releasing v7 stable.

File: tests/integration/generate_fixtures/go_ethereum.py [hunk 2]
```diff
@@ -104,8 +100,8 @@ def get_geth_process(geth_binary, datadir, geth_port):
     popen_proc = subprocess.Popen(
         run_geth_command,
         stdin=subprocess.PIPE,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
+        # stdout=subprocess.PIPE,
```
Comment (kclowes) [inline tests/integration/generate_fixtures/go_ethereum.py]: Should these go? Or be uncommented?

File: tests/integration/generate_fixtures/go_ethereum.py [hunk 3]
```diff
@@ -115,8 +111,8 @@ def get_geth_process(geth_binary, datadir, geth_port):
 
     print(
         "Geth Process Exited:\n"
-        f"stdout:{to_text(output)}\n\n"
-        f"stderr:{to_text(errors)}\n\n"
+        # f"stdout:{to_text(output)}\n\n"
```
Comment (kclowes) [inline tests/integration/generate_fixtures/go_ethereum.py]: Same here - can these go or should they be put back?

**Review by kclowes — APPROVED** [review]
LGTM! I left a few inline nits, and then the only other thing is I wonder if we should think of a better name for `accounts[0]` to reference it by. Maybe `default_account`? Feel free to take or leave, I don't think it's a huge improvement and I don't really have a better idea at the moment ¯\_(ツ)_/¯

File: tests/integration/generate_fixtures/go_ethereum.py (hunk 2 above)
Comment (fselmo) [reply tests/integration/generate_fixtures/go_ethereum.py]: Yes this was in order to get better logs, thank you!

**Comment by fselmo** [comment]
> LGTM! I left a few inline nits, and then the only other thing is I wonder if we should think of a better name for `accounts[0]` to reference it by. Maybe `default_account`? Feel free to take or leave, I don't think it's a huge improvement and I don't really have a better idea at the moment ¯_(ツ)_/¯

Yeah, agreed. I think that might be a nice pytest fixture abstraction to have if anything. I'll take a peek while I'm in there.

**Review by reedsa — APPROVED** [review]

**Comment by fselmo** [comment]
@kclowes, I opted to make use of the `w3.eth.default_account`, [in the most recent commit](https://github.com/ethereum/web3.py/pull/3307/commits/46a717a63b88bd19a0e249bf4546f31f7a17849b), by setting that to the first account for tests. I think that makes the most sense, does that feel like a good resolution to you? I added a test that makes sure `w3.eth.default_account` is empty by default since that was being checked in another test.

**Review by kclowes — APPROVED** [review]
Yeah, I like that! 🚢

**Pull request merged**